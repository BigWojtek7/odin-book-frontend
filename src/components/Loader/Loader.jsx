import styles from './Loader.module.css'
function Loader(){
  return(
    <div className={styles.loader}>
    <p className={styles.loading}>Data is loading...</p>
    <p className={styles.info}>Loading the first time may take longer - the server is waking up</p>
    <div className={styles.spinner}></div>
    </div>
  )
}

export default Loader