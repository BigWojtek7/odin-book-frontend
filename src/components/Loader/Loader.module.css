.loader {
  margin-top: 3em;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1em;
}
.loading {
  font-size: 2rem;
}
.loader .spinner {
  border: 16px solid #f3f3f3;
  border-top: 16px solid #0284c7;
  border-radius: 50%;
  width: 120px;
  height: 120px;
  animation: spin 3s linear infinite;
}

.info {
  font-size: 1rem;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
