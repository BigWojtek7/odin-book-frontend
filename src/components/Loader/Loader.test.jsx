import { render, screen } from '@testing-library/react';

import Loader from './Loader';

it('renders Loader component', () => {
  const { container } = render(<Loader />);

  expect(container).toMatchSnapshot();
});
