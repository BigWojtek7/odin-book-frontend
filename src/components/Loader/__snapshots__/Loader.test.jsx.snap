// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render Loader component 1`] = `
<div>
  <div
    class="_loader_bf668d"
  >
    <p
      class="_loading_bf668d"
    >
      Data is loading...
    </p>
    <p
      class="_info_bf668d"
    >
      Loading the first time may take longer - the server is waking up
    </p>
    <div
      class="_spinner_bf668d"
    />
  </div>
</div>
`;

exports[`renders Loader component 1`] = `
<div>
  <div
    class="_loader_bf668d"
  >
    <p
      class="_loading_bf668d"
    >
      Data is loading...
    </p>
    <p
      class="_info_bf668d"
    >
      Loading the first time may take longer - the server is waking up
    </p>
    <div
      class="_spinner_bf668d"
    />
  </div>
</div>
`;
